dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/trackedentity/OrderAndFilterTrackedEntityChangeLogTest.java

```diff
@@ -163,18 +163,24 @@ void shouldSortChangeLogsWhenOrderingByAttributeAsc()
     String updatedValue = "100";
 
     updateAttributeValue(trackedEntity, trackedEntityAttribute, updatedValue);
+    updateAttributeValue(trackedEntity, "dIVt4l5vIOa", "new value");
 
-    Page<TrackedEntityChangeLog> changeLogs =
-        trackedEntityChangeLogService.getTrackedEntityChangeLog(
-            UID.of(trackedEntity), null, params, defaultPageParams);
-
-    assertNumberOfChanges(3, changeLogs.getItems());
+    List<String> changeLogs =
+        trackedEntityChangeLogService
+            .getTrackedEntityChangeLog(UID.of(trackedEntity), null, params, defaultPageParams)
+            .getItems()
+            .stream()
+            .map(this::getDisplayName)
+            .toList();
 
-    assertAll(
-        () ->
-            assertUpdate(trackedEntityAttribute, "88", updatedValue, changeLogs.getItems().get(0)),
-        () -> assertCreate(trackedEntityAttribute, "88", changeLogs.getItems().get(1)),
-        () -> assertCreate("toUpdate000", "summer day", changeLogs.getItems().get(2)));
+    assertEquals(
+        List.of(
+            "numeric-attribute",
```

Comment:
I wonder if there is a way to more clearly tie what you expect to the setup. Before it was easier for me to see that you expect update, create, ... via the assertions